langchain4j-core/src/test/java/dev/langchain4j/model/chat/common/StreamingChatLanguageModelIT.java

```diff
@@ -0,0 +1,228 @@
+package dev.langchain4j.model.chat.common;
+
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.model.chat.StreamingChatLanguageModel;
+import dev.langchain4j.model.chat.request.ChatRequest;
+import dev.langchain4j.model.chat.request.json.JsonObjectSchema;
+import dev.langchain4j.model.chat.response.ChatResponse;
+import dev.langchain4j.model.chat.response.StreamingChatResponseHandler;
+import dev.langchain4j.model.output.TokenUsage;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.condition.EnabledIf;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static dev.langchain4j.model.output.FinishReason.STOP;
+import static dev.langchain4j.model.output.FinishReason.TOOL_EXECUTION;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.mockito.Mockito.spy;
+
+/**
+ * This test makes sure that all {@link StreamingChatLanguageModel} implementations behave consistently.
+ * <p>
+ * Make sure these dependencies are present in the module where this test class is extended:
+ * <pre>
+ *
+ * <dependency>
+ *     <groupId>dev.langchain4j</groupId>
+ *     <artifactId>langchain4j-core</artifactId>
+ *     <scope>test</scope>
+ * </dependency>
+ *
+ * <dependency>
+ *     <groupId>dev.langchain4j</groupId>
+ *     <artifactId>langchain4j-core</artifactId>
+ *     <classifier>tests</classifier>
+ *     <type>test-jar</type>
+ *     <scope>test</scope>
+ * </dependency>
+ *
+ * <dependency>
+ *     <groupId>org.mockito</groupId>
+ *     <artifactId>mockito-core</artifactId>
+ *     <scope>test</scope>
+ * </dependency>
+ *
+ * <dependency>
+ *     <groupId>org.mockito</groupId>
+ *     <artifactId>mockito-junit-jupiter</artifactId>
+ *     <scope>test</scope>
+ * </dependency>
+ */
+@TestInstance(PER_CLASS)
+public abstract class StreamingChatLanguageModelIT {
```

Comment:
```suggestion
public abstract class AbstractStreamingChatLanguageModelIT {
```

Reply:
@langchain4j , kudos for introducing abstract test classes, to avoid code duplication! 👍🏻